.tugboat/config.yml

```diff
@@ -1,55 +1,95 @@
+                - cp ${TUGBOAT_ROOT}/.tugboat/.htaccess ${DOCROOT}/.htaccess
 
                 # Update permalinks to remove the index.php.
                 - wp --allow-root --path="${DOCROOT}" option set permalink_structure /%postname%/
                 - wp --allow-root --path="${DOCROOT}" rewrite flush --hard
 
+            # Phase 2 (update): Import files, database, or any other assets that your
+            # website needs to run.
+            # When you refresh a Tugboat Preview, the process starts here, skipping `init`.
             update:
-
                 # Import uploads
                 - tar -C /tmp -zxf uploads.tar.gz
-                - rsync -av --delete /tmp/uploads/ "${TUGBOAT_ROOT}/docroot/wp-content/uploads/"
+                - rsync -avz --delete /tmp/uploads/ "${TUGBOAT_ROOT}/docroot/wp-content/uploads/"
+                - chgrp -R www-data "${DOCROOT}/wp-content/uploads"
+                - find "${DOCROOT}/wp-content/uploads" -type d -exec chmod 2775 {} \;
+                - find "${DOCROOT}/wp-content/uploads" -type f -exec chmod 0664 {} \;
 
                 # Cleanup
                 - apt-get clean
                 - rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
+            # These commands rewrite the hostname set in the database to use the current preview hostname.
+            #build:
+                #- wp --allow-root --path="${DOCROOT}" search-replace 'main-hbpfbyk632i25rgdo9caeoo9oxs4mrtd.tugboat.qa' "${TUGBOAT_SERVICE_URL_HOST}" --skip-columns=guid
+                # wp --allow-root --path=/var/lib/tugboat/docroot search-replace 'main-hbpfbyk632i25rgdo9caeoo9oxs4mrtd.tugboat.qa' 'pr78-vvg1syntykx8rfmnpy8dpo6hgjxvxiic.tugboatqa.com' --skip-columns=guid
+
+            # If this preview was built from a base preview, then the base preview's URLs will be set in the database,
+            # so we want to rewrite the base preview's URL to this preview's.  However, if this preview does NOT have a
+            # base preview, then the original URL from the database export will need to be rewritten.
             build: |
                 if [ "x${TUGBOAT_BASE_PREVIEW}" != "x" ]; then
                     wp --allow-root --path="${DOCROOT}" search-replace "${TUGBOAT_BASE_PREVIEW_URL_HOST}" "${TUGBOAT_SERVICE_URL_HOST}" --skip-columns=guid
+                    echo "Replacing ${TUGBOAT_BASE_PREVIEW_URL_HOST} with ${TUGBOAT_SERVICE_URL_HOST}"
                 else
-                    wp --allow-root --path="${DOCROOT}" search-replace 'main-hbpfbyk632i25rgdo9caeoo9oxs4mrtd.tugboat.qa' "${TUGBOAT_SERVICE_URL_HOST}" --skip-columns=guid
+                    wp --allow-root --path="${DOCROOT}" search-replace 'pr78-vvg1syntykx8rfmnpy8dpo6hgjxvxiic.tugboatqa.com' "${TUGBOAT_SERVICE_URL_HOST}" --skip-columns=guid
+                    echo "Replacing pr78-vvg1syntykx8rfmnpy8dpo6hgjxvxiic.tugboatqa.com with ${TUGBOAT_SERVICE_URL_HOST}"
```

Comment:
Has the database been modified in this PR? If not, these lines should not change.

Reply:
Yes, I used the preview to dump a new database with "The Tug Life" content removed and any necessary database updates run that I could see.